### PR TITLE
Fix clean for pattern subscriptions

### PR DESCRIPTION
**Pull request by philipp94831**

### CODE REVIEW

File: src/main/java/com/bakdata/kafka/util/TopicSubscription.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+package com.bakdata.kafka.util;
+
+import java.util.Collection;
+
+@FunctionalInterface
+interface TopicSubscription {
+
+    Collection<String> resolveTopics(Collection<String> allTopics);
```
Comment (torbsto) [inline src/main/java/com/bakdata/kafka/util/TopicSubscription.java:32]:
```suggestion
    Collection<String> resolveTopics(final Collection<String> allTopics);
```

Comment (philipp94831) [reply src/main/java/com/bakdata/kafka/util/TopicSubscription.java:32]: As this is an interface, final does nothing here. Implementations can still do with the parameter what they are up to

File: src/test/java/com/bakdata/kafka/util/TopologyInformationTest.java [hunk 2]
```diff
@@ -120,4 +122,16 @@ void shouldReturnAllPseudoInternalTopics() {
...
     }
 
+    @Test
+    void shouldResolvePatternTopics() {
+        final StreamsBuilder streamsBuilder = new StreamsBuilder();
+        final KStream<String, Object> stream = streamsBuilder.stream(Pattern.compile(".*-topic"));
+        stream.to("output");
+        final TopologyInformation topologyInformation = new TopologyInformation(streamsBuilder.build(), "id");
+        assertThat(
+                topologyInformation.getExternalSourceTopics(List.of("foo", "foo-topic", "foo-topic-bar", "bar-topic")))
```
[the remark this answers is not in the document]
Comment (philipp94831) [reply src/test/java/com/bakdata/kafka/util/TopologyInformationTest.java:132]: Added, thanks

File: src/test/java/com/bakdata/kafka/integration/StreamsCleanUpTest.java [hunk 3]
```diff
@@ -316,6 +317,23 @@ void shouldReprocessAlreadySeenRecords() throws InterruptedException {
         this.runAndAssertSize(6);
     }
 
+    @Test
+    void shouldReprocessAlreadySeenRecordsWithPattern() throws InterruptedException {
+        this.app = this.createWordCountPatternApplication();
+        final SendValuesTransactional<String> sendRequest =
```
[the remark this answers is not in the document]
Comment (philipp94831) [reply src/test/java/com/bakdata/kafka/integration/StreamsCleanUpTest.java:323]: Good idea, added

**Pull request merged**